File: backend/app/schemas/workspace.py
from pydantic import BaseModel, Field
from typing import Optional
from enum import IntEnum
from .base import AuditSchema, SoftDeleteSchema
from .user import UserLite, UserResponse

class WorkspaceRole(IntEnum):
    ADMIN = 20
    MEMBER = 15
    GUEST = 5

class WorkspaceBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=255)
    slug: str = Field(..., min_length=1, max_length=50, pattern=r'^[a-z0-9-]+$')
    organization_size: Optional[str] = None
    timezone: str = "UTC"

class WorkspaceCreate(WorkspaceBase):
    pass

class WorkspaceUpdate(BaseModel):
    name: Optional[str] = None
    logo_url: Optional[str] = None
    timezone: Optional[str] = None

class WorkspaceResponse(AuditSchema, SoftDeleteSchema, WorkspaceBase):
    logo_url: Optional[str] = None
    owner_id: int

class WorkspaceLite(BaseModel):
    id: int
    name: str
    slug: str

class WorkspaceMemberBase(BaseModel):
    role: WorkspaceRole

class WorkspaceMemberCreate(WorkspaceMemberBase):
    member_id: int

class WorkspaceMemberResponse(AuditSchema, WorkspaceMemberBase):
    member: UserLite
    is_active: bool
    joining_date: Optional[str] = None

File: backend/internal/dto/response/workspace.go
package response

import "time"

// WorkspaceResponse is the full workspace representation.
type WorkspaceResponse struct {
	ID               uint64         `json:"id"`
	Name             string         `json:"name"`
	Slug             string         `json:"slug"`
	LogoURL          *string        `json:"logo_url"`
	OrganizationSize *string        `json:"organization_size"`
	Timezone         string         `json:"timezone"`
	OwnerID          uint64         `json:"owner_id"`
	Owner            UserResponse   `json:"owner"`
	TotalMembers     int64          `json:"total_members"`
	TotalProjects    int64          `json:"total_projects"`
	CreatedAt        time.Time      `json:"created_at"`
	UpdatedAt        time.Time      `json:"updated_at"`
	CreatedByID      *uint64        `json:"created_by_id"`
	UpdatedByID      *uint64        `json:"updated_by_id"`
	DeletedAt        *time.Time     `json:"deleted_at"`
	IsDeleted        bool           `json:"is_deleted"`
}

// WorkspaceLite is a compact workspace representation.
type WorkspaceLite struct {
	ID   uint64 `json:"id"`
	Name string `json:"name"`
	Slug string `json:"slug"`
}

// WorkspaceMemberResponse represents a workspace member.
type WorkspaceMemberResponse struct {
	ID          uint64    `json:"id"`
	WorkspaceID uint64    `json:"workspace_id"`
	UserID      uint64    `json:"user_id"`
	Role        int       `json:"role"`
	IsActive    bool      `json:"is_active"`
	User        UserLite  `json:"user"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}
